src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -121,7 +165,10 @@ class SRAMInterface[T <: Data](
   private[chisel3] var _underlying: Option[HasTarget] = None
 
   /** Target information for annotating the underlying SRAM if it is known. */
-  def underlying: Option[HasTarget] = _underlying
+  def underlying:                    Option[HasTarget] = _underlying
+  private val descriptionDefinition: Definition[SRAMDescription] = Instantiate.definition(new SRAMDescription)
+  private val descriptionType:       ClassType = descriptionDefinition.getClassType
+  val description:                   Property[ClassType] = Property[descriptionType.Type]()
```

Comment:
The `Definition` should not be a field of the `Bundle`, it should possibly be a field of the companion object (if we're saying there's only 1 definition for all `SRAMDescriptions`), or possibly an argument to the constructor. I'm thinking maybe we should go with argument to constructor because I suspect users are going to want to optionally add their own metadata to SRAMs which implies there may not be a single definition for all of them.

One thing to note is this is a very backwards incompatible change, anyone using 

We also want the description to be typed, `Property[SRAMDescription]`, otherwise this is weird and erased and not useful as a Chisel API (e.g. our use case of accessing the `hierarchy` field from it).

Reply:
> it should possibly be a field of the companion object

Sounds good, I will do it later. For user adding their own metadata, Explicitly instantiating the `SRAMDescription` and connect sounds dangerous, I think we should consider what to add and how to add. e.g. SRAM description string, Initial ROM, UPF fields, etc. 

> accessing the hierarchy field from it

yes if you need to access the `hierarchy` field, we need add new API to `Class` to extract it.